src/Plugins/Cache.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Peck\Plugins;
+
+final readonly class Cache
+{
+    public function __construct(
+        private string $cacheDirectory,
+    ) {
+        if (! is_dir($this->cacheDirectory)) {
+            mkdir($this->cacheDirectory, 0777, true);
+        }
+    }
+
+    /**
+     * Creates the default instance of Spellchecker.
+     */
+    public static function default(): self
+    {
+        return new self(__DIR__.'/../../.peck.cache');
+    }
+
+    public function get(string $key): mixed
+    {
+        $cacheFile = $this->getCacheFile($key);
+
+        if (! file_exists($cacheFile)) {
+            return null;
+        }
+
+        /**
+         * @var string
+         */
+        $serializedContents = file_get_contents($cacheFile);
```

Comment:
The function file_get_functions ([Docs](https://www.php.net/manual/en/function.file-get-contents.php)) could return `false`. In this case we should return null instead of unserializing the data. Another Idea would be to throw an exception, but that might be a bit too much. In a later iteration we could throw an exception, catch it somewhere and then log this.


```suggestion
        $serializedContents = file_get_contents($cacheFile);
        
        if ($serializedContents === null) {
                return null;
        }
```